LanguageFeatures/Inline-classes/dynamic_semantics_inline_class_A03_t01.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A type test, o is U or o is! U, and a type cast, o as U, where U
+/// is or contains an inline type, is performed at run time as a type test and
+/// type cast on the run-time representation of the inline type
+///
+/// @description Check that at run time a type test, `o is U` or `o is! U` is
+/// performed as a type test on the run-time representation of the inline type
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+inline class V {
+  final int id;
+  V(this.id);
+}
+
+main() {
+  var v = V(42);
+  Expect.isTrue(v is V);
```

Comment:
This one is a little bit tricky. Here's what I'm thinking about as a natural way to get good coverage:

What we want to show is that `_ is V` works the same as `_ is int`, no matter what we put into the location where we have `_`. So maybe we could do this:

```dart
var objects = [true, 1, 1.5, #foo, <String>[], {'a': () {}}, StringBuffer('Hello')]; // A bunch of objects.

void main() {
  for (var o in objects) {
    Expect.equals(o is V, o is int);
    Expect.equals(o is! V, o is! int);
  }
  ...
}
```

Perhaps it doesn't make much difference, but it does put more emphasis on "`_ is V` works the same as `_ is int`", whereas only asking about these expressions with `v` allows those two expressions to behave in different ways for any other type than `V`, and we'd want to catch that.

Reply:
Great, thanks!